pkg/crc/preflight/preflight_checks_nonlinux.go

```diff
@@ -72,3 +75,42 @@ func waitForDaemonRunning() error {
 		return nil
 	}, 2*time.Second)
 }
+
+func sshPortCheck() Check {
+	return Check{
+		configKeySuffix:  "check-ssh-port",
+		checkDescription: "Checking SSH port availability",
+		check:            checkSSHPortFree(),
+		fixDescription:   fmt.Sprintf("crc uses %d to run SSH", constants.VsockSSHPort),
+		flags:            NoFix,
+		labels:           None,
+	}
+}
+
+func checkSSHPortFree() func() error {
+	return func() error {
+
+		host := net.JoinHostPort(constants.LocalIP, strconv.Itoa(constants.VsockSSHPort))
+
+		daemonClient := daemonclient.New()
+		exposed, err := daemonClient.NetworkClient.List()
+		if err == nil {
+			// if port already exported by vsock we could proceed
+			for _, e := range exposed {
+				if e.Local == host {
+					return nil
+				}
+			}
+		}
+
+		server, err := net.Listen("tcp", host)
+		// if it fails then the port is likely taken
+		if err != nil {
+			return fmt.Errorf("port %d already in use: %s", constants.VsockSSHPort, err)
+		}
+		server.Close()
+
+		// we successfully used and closed the port
+		return nil
```

Comment:
```suggestion
		return err
```

Reply:
If the intent is to return the error from the deferred `server.Close()` call, I don't think this will work as expected:
https://go.dev/play/p/HyKboZE2dka

Reply:
You are right. We can define the err variable as part of the function signature to set the correct return value in defer.
https://go.dev/play/p/8kaB3HQlfk6